### PR TITLE
Fix fastparquet tests to work with HDFS

### LINKED ISSUE #9545: [BUG] Dataproc 2.0 test_reading_file_rewritten_with_fastparquet tests failing
**Describe the bug**
Our integration tests on Dataproc 2.0, the test_reading_file_rewritten_with_fastparquet tests are failing:

```
2023-10-25T16:07:03.534Z] FAILED ../../src/main/python/fastparquet_compatibility_test.py::test_reading_file_written_with_fastparquet[Byte(not_null)][INJECT_OOM] - FileNotFoundError: [Errno 2] No such fil
e or directory: '/tmp/pyspark_tests...
[2023-10-25T16:07:03.534Z] FAILED ../../src/main/python/fastparquet_compatibility_test.py::test_reading_file_written_with_fastparquet[Byte][INJECT_OOM] - FileNotFoundError: [Errno 2] No such file or direc
tory: '/tmp/pyspark_tests...
[2023-10-25T16:07:03.534Z] FAILED ../../src/main/python/fastparquet_compatibility_test.py::test_reading_file_written_with_fastparquet[Short(not_null)][INJECT_OOM] - FileNotFoundError: [Errno 2] No such fi
le or directory: '/tmp/pyspark_tests...
[2023-10-25T16:07:03.534Z] FAILED ../../src/main/python/fastparquet_compatibility_test.py::test_reading_file_written_with_fastparquet[Short][INJECT_OOM] - FileNotFoundError: [Errno 2] No such file or dire
ctory: '/tmp/pyspark_tests...
[2023-10-25T16:07:03.534Z] FAILED ../../src/main/python/fastparquet_compatibility_test.py::test_reading_file_written_with_fastparquet[Integer(not_null)][INJECT_OOM] - FileNotFoundError: [Errno 2] No such 
file or directory: '/tmp/pyspark_tests...
[2023-10-25T16:07:03.534Z] FAILED ../../src/main/python/fastparquet_compatibility_test.py::test_reading_file_written_with_fastparquet[Integer][INJECT_OOM] - FileNotFoundError: [Errno 2] No such file or di
rectory: '/tmp/pyspark_tests...
[2023-10-25T16:07:03.534Z] FAILED ../../src/main/python/fastparquet_compatibility_test.py::test_reading_file_written_with_fastparquet[Long(not_null)] - FileNotFoundError: [Errno 2] No such file or directo
ry: '/tmp/pyspark_tests...
[2023-10-25T16:07:03.534Z] FAILED ../../src/main/python/fastparquet_compatibility_test.py::test_reading_file_written_with_fastparquet[Long][INJECT_OOM] - FileNotFoundError: [Errno 2] No such file or direc
tory: '/tmp/pyspark_tests...
[2023-10-25T16:07:03.534Z] FAILED ../../src/main/python/fastparquet_compatibility_test.py::test_reading_file_written_with_fastparquet[Float(not_null)][INJECT_OOM] - FileNotFoundError: [Errno 2] No such fi
le or directory: '/tmp/pyspark_tests...
[2023-10-25T16:07:03.534Z] FAILED ../../src/main/python/fastparquet_compatibility_test.py::test_reading_file_written_with_fastparquet[Float] - FileNotFoundError: [Errno 2] No such file or directory: '/tmp
/pyspark_tests...
[2023-10-25T16:07:03.534Z] FAILED ../../src/main/python/fastparquet_compatibility_test.py::test_reading_file_written_with_fastparquet[Double(not_null)] - FileNotFoundError: [Errno 2] No such file or direc
tory: '/tmp/pyspark_tests...
[2023-10-25T16:07:03.534Z] FAILED ../../src/main/python/fastparquet_compatibility_test.py::test_reading_file_written_with_fastparquet[Double] - FileNotFoundError: [Errno 2] No such file or directory: '/tm
p/pyspark_tests...
[2023-10-25T16:07:03.534Z] FAILED ../../src/main/python/fastparquet_compatibility_test.py::test_reading_file_written_with_fastparquet[Decimal(not_null)(18,0)][INJECT_OOM] - FileNotFoundError: [Errno 2] No
 such file or directory: '/tmp/pyspark_tests...
[2023-10-25T16:07:03.534Z] FAILED ../../src/main/python/fastparquet_compatibility_test.py::test_reading_file_written_with_fastparquet[Decimal(18,0)][INJECT_OOM] - FileNotFoundError: [Errno 2] No such file
 or directory: '/tmp/pyspark_tests...
[2023-10-25T16:07:03.534Z] FAILED ../../src/main/python/fastparquet_compatibility_test.py::test_reading_file_rewritten_with_fastparquet[Date(not_null)-int961] - FileNotFoundError: [Errno 2] No such file o
r directory: '/tmp/pyspark_tests...
[2023-10-25T16:07:03.534Z] FAILED ../../src/main/python/fastparquet_compatibility_test.py::test_reading_file_rewritten_with_fastparquet[Date-int96] - FileNotFoundError: [Errno 2] No such file or directory
: '/tmp/pyspark_tests...
[2023-10-25T16:07:03.534Z] FAILED ../../src/main/python/fastparquet_compatibility_test.py::test_reading_file_rewritten_with_fastparquet[Timestamp(not_null)-int960][INJECT_OOM] - FileNotFoundError: [Errno 
2] No such file or directory: '/tmp/pyspark_tests...
[2023-10-25T16:07:03.534Z] FAILED ../../src/main/python/fastparquet_compatibility_test.py::test_reading_file_rewritten_with_fastparquet[Timestamp-int96] - FileNotFoundError: [Errno 2] No such file or dire
ctory: '/tmp/pyspark_tests...
[2023-10-25T16:07:03.534Z] FAILED ../../src/main/python/fastparquet_compatibility_test.py::test_reading_file_rewritten_with_fastparquet[Struct(not_null)(('first', Integer(not_null)))-int96][INJECT_OOM] - 
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/pyspark_tests...


```

Not sure why it has no such file, maybe there was another crash or failure that caused these?

### PR DESCRIPTION
**Pull request by mythrocks**
Fixes #9545.

This commit fixes the `fastparquet` tests to run on Spark clusters where the `fs.default.name` does not point to the local filesystem.

Before this commit, the `fastparquet` tests assumed that the parquet files generated for the tests were written to local filesystem, and could be read from both `fastparquet` and Spark from the same location.  However, this fails when run against clusters whose default filesystem is HDFS. `fastparquet` can only read from the local filesystem.

This commit changes the tests as follows:
1. For tests where data is generated by Spark, the data is copied to local filesystem before it is read by `fastparquet`.
2. For tests where data is generated by `fastparquet`, the data is copied to the default Hadoop filesystem before reading through Spark.

### CODE REVIEW

**Comment by mythrocks** [comment]
Build

File: integration_tests/src/main/python/fastparquet_compatibility_test.py [hunk 1]
```diff
@@ -167,6 +203,8 @@ def test_reading_file_written_with_gpu(spark_tmp_path, column_gen):
     There are xfails here because of fastparquet limitations in handling Decimal, Timestamps, Dates, etc.
     """
     data_path = spark_tmp_path + "/FASTPARQUET_TEST_GPU_WRITE_PATH"
+    local_base_path = (spark_tmp_path + "_local")
```
Comment (revans2) [inline integration_tests/src/main/python/fastparquet_compatibility_test.py:206]: I think this is okay, but it does assume that the temp path in HDFS is going to be available in the local file system too. I don't think there is anything we can do about that though.

**Review by revans2 — APPROVED** [review]

**Comment by mythrocks** [comment]
I've merged this change.  With any luck, this should unblock the CDH builds.

**Comment by gerashegalov** [comment]
Alternative:

fastparquet uses fsspec that provides a bunch of implementations out of the box https://github.com/fsspec/filesystem_spec/tree/master/fsspec/implementations . 
This includes WebHDFS that can be enabled on CDH.

**Pull request merged**